Userland/Applications/VideoPlayer/main.cpp

```diff
@@ -28,72 +45,98 @@ ErrorOr<int> serenity_main(Main::Arguments arguments)
+        //        by some color conversion library and then passed to something (GL?) for output.
+        auto const& output_y = vp9_decoder.get_output_buffer_for_plane(0);
+        auto const& output_u = vp9_decoder.get_output_buffer_for_plane(1);
+        auto const& output_v = vp9_decoder.get_output_buffer_for_plane(2);
+        auto y_size = vp9_decoder.get_y_plane_size();
+        auto uv_subsampling_y = vp9_decoder.get_uv_subsampling_y();
+        auto uv_subsampling_x = vp9_decoder.get_uv_subsampling_x();
+        Gfx::IntSize uv_size { y_size.width() >> uv_subsampling_x, y_size.height() >> uv_subsampling_y };
+
+        for (auto y_row = 0u; y_row < video_track.pixel_height; y_row++) {
+            auto uv_row = y_row >> uv_subsampling_y;
+
+            for (auto y_column = 0u; y_column < video_track.pixel_width; y_column++) {
+                auto uv_column = y_column >> uv_subsampling_x;
+
+                auto y = output_y[y_row * y_size.width() + y_column];
+                auto cb = output_u[uv_row * uv_size.width() + uv_column];
+                auto cr = output_v[uv_row * uv_size.width() + uv_column];
+                // Convert from Rec.709 YCbCr to RGB.
+                auto r_float = floorf(clamp(y + (cr - 128) * 219.0f / 224.0f * 1.5748f, 0, 255));
+                auto g_float = floorf(clamp(y + (cb - 128) * 219.0f / 224.0f * -0.0722f * 1.8556f / 0.7152f + (cr - 128) * 219.0f / 224.0f * -0.2126f * 1.5748f / 0.7152f, 0, 255));
+                auto b_float = floorf(clamp(y + (cb - 128) * 219.0f / 224.0f * 1.8556f, 0, 255));
+                auto r = static_cast<u8>(r_float);
+                auto g = static_cast<u8>(g_float);
+                auto b = static_cast<u8>(b_float);
+
+                image->set_pixel(y_column, y_row, Gfx::Color(r, g, b));
             }
         }
-    }
 
-display:
-    auto main_widget = TRY(window->try_set_main_widget<GUI::Widget>());
-    main_widget->set_fill_with_background_color(true);
-    main_widget->set_layout<GUI::VerticalBoxLayout>();
-    auto image_widget = TRY(main_widget->try_add<GUI::ImageWidget>());
-    image_widget->set_bitmap(image);
+        image_widget->set_bitmap(image);
+        image_widget->update();
+
+        frame_index++;
+        frame_number++;
+    };
+
     image_widget->set_fixed_size(video_track.pixel_width, video_track.pixel_height);
+    image_widget->on_click = [&]() { display_next_frame(); };
+
+    if (benchmark) {
+        timeval start, end;
+        gettimeofday(&start, nullptr);
+        for (auto i = 0; i < 100; i++)
+            display_next_frame();
+        gettimeofday(&end, nullptr);
+        auto seconds = end.tv_sec - start.tv_sec;
+        auto useconds = end.tv_usec - start.tv_usec;
+        auto total_milliseconds = (seconds * 1000) + (useconds / 1000);
+        outln("Decoding 100 frames took {} ms", total_milliseconds);
+        return 0;
+    }
```

Comment:
You could use `ElapsedTimer` from `LibCore` instead of doing this the manual way with libc.

Reply:
Fixed.